src/cmd/singularity/cli/actions.go

```diff
@@ -274,6 +282,14 @@ func execWrapper(cobraCmd *cobra.Command, image string, args []string, name stri
 		}
 	}
 
+	// Transpose environment
+	for _, e := range environment {
```

Comment:
Not sure why this is being done in two different loops, this adds unnecessary complexity